Fix booleans returned when setting variants

ajrb opened the pull request:
The return values were all inverted erroneously, rather embarrassing. Not seen anyone using these return values yet, so should cause no issues.

KABoissonneault commented:
Wouldn't it be safer to just update the comments to "True if this X had no existing variants", and changing the variable name? It's not like either way is more intuitive for this function, I think as long as the docs match the behavior, it should be good

ajrb commented:
It was wrong and I fixed it. I don't feel it needs the API changing for safety however if that's the consensus I will change it.

ajrb commented:
I've done as requested. 
Added method documentation for getters.
Added new block variant getter for when not iterating locations. (i.e. most mod usage)

KABoissonneault commented on Assets/Scripts/Utility/AssetInjection/WorldDataVariants.cs:
```diff
@@ -218,6 +235,28 @@ public static string GetBuildingVariant(ref BlockRecordKey blockRecordKey, strin
         }
 
+        /// <summary>
+        /// Gets a variant set for the specified building, based on supplied region and location.
```
>Gets a variant set for the specified building"
Should that be "block"?

ajrb replied:
Yep, thanks for the catch.
